src/api/index.ts

```diff
@@ -40,8 +43,91 @@ export class JiraAPI {
     this.proxyPath = options.proxyPath ?? DEFAULT_PROXY_PATH;
   }
 
+  private generateProjectUrl = (url: string) => new URL(url).origin;
+
   private async getApiUrl() {
     const proxyUrl = await this.discoveryApi.getBaseUrl('proxy');
     return `${proxyUrl}${this.proxyPath}`;
   }
+
+  private convertToString = (arrayElement: Array<string>): string =>
+    arrayElement
+    .filter(Boolean)
+    .map(i => `'${i}'`).join(',');
+
+  private async getIssuesCountByType(
+    apiUrl: string,
+    projectKey: string,
+    component: string,
+    statusesNames: Array<string>,
+    issueType: string,
+    issueIcon: string
+  ) {
+    const statusesString = this.convertToString(statusesNames);
+    const jql = `
+      project = "${projectKey}"
+      AND issuetype = "${issueType}"
```

Comment:
I suppose it's not possible to query like so `issueType in (...)` ?

Reply:
Actually it is possible but I think operator ```in``` is for multiple options field for example status. I take it directly from JQL generated by Jira itself.

Reply:
FWIW, In Jira Query Language (JQL), `IN` can apply to most types.  In this example, `issueType = "Epic"` is equivalent to `issueType in ("Epic")`.

Reply:
The reason I was asking is that I was hoping we could avoid running multiple JQL queries by returning all the relevant issues. We'd have to calculate counts by issueType client side (unless I'm missing something in jql) but I think that's preferable to running multiple jql queries (in my experience jira can be veeeerrrry slow)

Reply:
Amazing idea, I will definitely do that

Reply:
Sadly that was premature, when you create JQL for multiple types it won't return you total number of issues by type. So it won't be possible to merge all of this requests into one.

Reply:
Right! As @iain-b points out, that would need to be calculated client side. So if there were 5 epics, 10 stories, and 3 bugs, your returned issue could would be 18 and you'd need to client-side loop through your result set to do counts.

Reply:
The problem with that approach is that Jira limits number of elements in response to 100, so for example for a project with 3000 issues I would need to made 30 requests to fetch all of them, add all of them to a single array and then iterate over an array and count issues per type. Using single issueType per query return total number of issues without any limitations. I tested it on my example Jira project. 

To make it more visible lets compare responses size for two types of approaches.
1. Fetch each issueType separately (258 issues in total):
![uAWEpU8](https://user-images.githubusercontent.com/36006588/99702687-6dc4b100-2a96-11eb-87df-2a71724c6d3c.png)
2. Fetch all issuesType together (only first 100 of them, we would need to make 2 more similar requests in order to get all issues):
![zn9GyGk](https://user-images.githubusercontent.com/36006588/99702943-c7c57680-2a96-11eb-8575-4148e95d2ec2.png)

Thanks for pointing this out tho, that was really good idea but sadly Jira REST API is not that friendly as it could be.